understanding/20/link-purpose-in-context.html

```diff
@@ -1,403 +1,187 @@
+      <p>A best practice for links to <a>conforming alternate versions</a> is to ensure that the link text to the
+         conforming alternate version indicates in link text that the page it leads to represents the more accessible
+         version. This information may also be provided in text - the goal is to ensure that the end user knows what the
+         purpose of the link is.</p>
       <p>The Success Criterion includes an exception for links for which the purpose of the
          link cannot be determined from the information on the Web page. In this situation,
          the person with the disability is not at a disadvantage; there is no additional context
          available to understand the link purpose. However, whatever amount of context is available
          on the Web page that can be used to interpret the purpose of the link must be made
          available in the link text or programmatically associated with the link to satisfy
-         the Success Criterion.
-      </p>
-      
+         the Success Criterion.</p>
       <div class="note">
-         
          <p>There may be situations where the purpose of the link is is supposed to be unknown
             or obscured. For instance, a game may have links identified only as door #1, door
             #2, and door #3. This link text would be sufficient because the purpose of the links
-            is to create suspense for all users.
-         </p>
-         
+            is to create suspense for all users.</p>
       </div>
-      
-      <p>See also 
-         <a href="link-purpose-link-only">2.4.9: Link Purpose (Link Only)</a>.
-      </p>
-      
-      
+      <p>As <q>link text together with its <a>programmatically determined link context</a></q> may
+         only benefit assistive technology users, see also <a href="link-purpose-link-only">2.4.9 Link Purpose (Link Only)</a>,
+         which only relies on the link text itself.</p>
    </section>
    <section id="benefits">
       <h2>Benefits of Link Purpose (In Context)</h2>
-      
-      
       <ul>
-         
-         <li>This Success Criterion helps people with motion impairment by letting them skip links
-            that they are not interested in, avoiding the keystrokes needed to visit the referenced
-            content and then returning to the current content.
-         </li>
-         
-         <li>People with cognitive limitations will not become disoriented by multiple means of
-            navigation to and from content they are not interested in.
-         </li>
-         
-         <li>
-            People with visual disabilities will be able to determine the purpose of a link by
-            exploring the link's context. 
-            
-            
-         </li>
-         
+         <li>Links whose link text clearly conveys their purpose can help people who navigate with a
+            keyboard or keyboard-like interface, letting them skip links that they are not interested in,
```

Comment:
this seems more like rational that helps everyone rather than just people using keyboard?   As a person who navigates using [ pointer, keyboard, touch, voice dictation, etc. ] if link text is descriptive, I can ignore links that aren't of interest to me.

Reply:
additionally, this seems to stress on link-text alone being descriptive enough (2.4.9), but being this is link purpose in context, why isn't this at least also making mention of a link and its surrounding text?

Reply:
to satisfy the SC you have link text OR link text plus programmatic context (which may not necessarily be surrounding text, but also things like `aria-describedby` pointing somewhere completely differently). so I was trying to make the distinction here for "if you satisfy the SC with the first bit, that'll help all/most people (even without AT), whereas the programmatic bit may only help people WITH AT"

Reply:
yeah, i get that regarding its link text OR link text + context for the SC.   I think i just globed onto it being perceived more as a 2.4.9 recommendation because I was already questioning rational of the benefit specific to keyboard users that doesn't equally help other people with/without disabilities.  

so never mind then the comment about being more 2.4.9 focused.  But I still question this specific benefit callout in general.

Reply:
@scottaohara following your (very valid, I hasten to add) points, I've had another go at tweaking the benefits. let me know what you think